include/rocksdb/filter_policy.h

```diff
@@ -65,6 +65,37 @@ class FilterBitsBuilder {
   // The ownership of actual data is set to buf
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) = 0;
 
+  // Similar to Finish(std::unique_ptr<const char[]>* buf), except that
+  // for a non-null status pointer argument, it will point to
+  // Status::Corruption() when there is any corruption during filter
+  // construction or Status::OK() otherwise.
+  //
+  // WARNING: do not use a filter resulted from a corrupted construction
+  virtual Slice Finish(std::unique_ptr<const char[]>* buf,
+                       Status* /* status */) {
+    return Finish(buf);
```

Comment:
Based on how the typical API signature, a signature returning a Status would be better:
  Status Finish(std::unique_ptr<...>, Slice *slice)

Reply:
I am reworking this signature in changes I am working on. (Also allowing a MemoryAllocator to be specified.)

Reply:
Sounds good!

Reply:
@pdillinger  Sounds good!